### PR TITLE
feat: profile references

### PR DESCRIPTION
**Pull request by tsaleksandrova**

### CODE REVIEW

File: docs/config/config.md [hunk 1]
```diff
@@ -9,11 +9,22 @@ If there are properties with the same name or ID on two config levels:
...
 * integration - default profile for E2E tests
 * visual - profile for visual tests
+* api - profile for API tests
+
+`profile` can also point to a file. The configuration in the referred file will have a lower priority.
+Profiles will be applied recursively from the referred file, its own profile reference and so on.
+Note that the paths will be relative to the UIVeri5 module. Use absolute paths for your project directory:
+```javascript
+exports.config = {
+  profile: path.join(__dirname, 'base/config.js'),
```
Comment (maximnaidenov) [inline docs/config/config.md]: this will require all existing tests to rewrite the profile reference or this is an extended option ?

Comment (tsaleksandrova) [reply docs/config/config.md]: extended, e.g. see the api profile and its e2e test

File: spec/configParser.spec.js [hunk 2]
```diff
@@ -202,4 +202,17 @@ describe("Should parse confkey from command-line", function () {
     ]);
   });
 
+  it('Should refer to another profile', function () {
```
Comment (maximnaidenov) [inline spec/configParser.spec.js]: can you please  add here (or move) the test that covers the simpler case "profile: integration" ?

File: src/configParser.js [hunk 3]
```diff
@@ -76,6 +79,26 @@ ConfigParser.prototype._mergeConfig = function (configFile, type) {
...
 
+ConfigParser.prototype._resolveProfile = function (profile, fullConfig) {
+  fullConfig = fullConfig || [];
+  if (!profile) {
+    return [];
+  }
+  var fileConfig;
+  try {
+    fileConfig = _readConfig(CONFIG_DIR + profile + '.profile.conf.js');
+  } catch (e) {
```
Comment (maximnaidenov) [inline src/configParser.js]: handling exceptions as part of normal processing is an antipattern IMHO.  I would personally return null for file-not-found and handle it respectively. Of course, any error will throw an exception.

Comment (tsaleksandrova) [reply src/configParser.js]: replaced this with a sync check returning a boolean.
unfortunately now the tests fail on ci, altough they are ok locally.
this is again not a good idea: it's strongly discouraged to check if a file exists, and then read from it, as the file may be deleted in the meantime. but all other options involve a try/catch

File: src/configParser.js [hunk 4]
```diff
@@ -76,6 +79,24 @@ ConfigParser.prototype._mergeConfig = function (configFile, type) {
   this.config = _mergeWithArrays(newConfig, priorityConfig);
 };
 
+ConfigParser.prototype._resolveProfile = function (profile, fullConfig) {
+  fullConfig = fullConfig || [];
+  if (!profile) {
+    return [];
+  }
+  var builtInFilePath = path.join(CONFIG_DIR, profile + '.profile.conf.js');
+  var pathToConfig = profile.indexOf(path.sep) > -1 ? profile : builtInFilePath;
```
Comment (maximnaidenov) [inline src/configParser.js]: this is a tricky choice - what if the user gives the simplest form of a relative path without separator - like "own.profile.conf.js"?. IMHO the readSync was good idea, we just needs a solution for the cwd vs installed dir problem. I was thinking of solving this by simply prepending the default files with __dirname and adding path.resolve(). This way require() will always get absolute path and no ambiguity will arise ?

**Pull request merged**